core/src/main/java/com/homeaway/streamplatform/streamregistry/StreamRegistryApplication.java

```diff
@@ -390,4 +414,67 @@ private void registerManagedContainer(final Environment environment, ManagedKStr
             new StreamRegistryManagedContainer(managedKStreams, managedInfraManager, managedKafkaProducer);
         environment.lifecycle().manage(managedContainer);
     }
+
+    private boolean isKafkaTopicAlreadyPresent(AdminClient adminClient, String topicName) throws InterruptedException, ExecutionException {
+        DescribeTopicsResult describeTopicsResult = adminClient.describeTopics(Collections.singleton(topicName));
+        try {
+            TopicDescription topicDescription = describeTopicsResult.values().get(topicName).get();
+            log.info("Topic present. {}", topicDescription);
+            return true;
+        } catch (ExecutionException e) {
+            if ("org.apache.kafka.common.errors.UnknownTopicOrPartitionException".equals(e.getCause().getClass().getName())) {
+                log.error("Topic {} not present.", topicName);
+                return false;
+            }
+            throw e;
+        }
+    }
+
+    private void makeSureTopicIsCompacted(AdminClient adminClient, String topicName) throws ExecutionException, InterruptedException {
+        ConfigResource configResource = new ConfigResource(ConfigResource.Type.TOPIC, topicName);
+
+        Config config = adminClient.describeConfigs(Collections.singleton(configResource)).values().get(configResource).get();
+        if (isCompactionConfigAvailable(config)) {
+            log.info("Compaction config available in the topic {}", topicName);
+        } else {
+            config.entries().add(new ConfigEntry("cleanup.policy", "compact"));
+            adminClient.alterConfigs(Collections.singletonMap(configResource, config))
+                    .values()
+                    .get(configResource)
+                    .get();
+            log.info("Stream Registry underlying data store - kafka topic {} updated with config {}", topicName, config);
+        }
+    }
+
+    private void createTopicWithCompactionEnabled(AdminClient adminClient, String topicName) throws ExecutionException, InterruptedException {
+        int partitions = 2;
+        short replicationFactor = 3;
+        NewTopic streamRegistryTopic = new NewTopic(topicName, partitions, replicationFactor);
+        Map<String, String> configs = new HashMap<>();
+        configs.put("min.insync.replicas", "2");
+        configs.put("cleanup.policy", "compact");
+        streamRegistryTopic.configs(configs);
+        try {
+            adminClient.createTopics(Collections.singleton(streamRegistryTopic)).values().get(topicName).get();
+        } catch (ExecutionException e) {
+            if ("org.apache.kafka.common.errors.InvalidReplicationFactorException".equals(e.getCause().getClass().getName())) {
+                log.warn("Some environments like dev or laptop may have a cluster with only one broker. So, try to create with one replica.");
+                partitions = 1;
```

Comment:
In config-dev.yaml set these these variables to 1 so you may not need this duplication.